components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -236,6 +246,28 @@ def get_clp_connection_params_and_type(self, disable_localhost_socket_connection
             connection_params_and_type["ssl_cert"] = self.ssl_cert
         return connection_params_and_type
 
+    def dump_to_primitive_dict(self):
+        return self.dict(exclude={"username", "password"})
+
+    def load_credentials_from_file(self, credentials_file_path: pathlib.Path):
+        config = read_yaml_config_file(credentials_file_path)
+        if config is None:
+            raise ValueError(f"Credentials file '{credentials_file_path}' is empty.")
+        try:
+            self.username = get_config_value(config, f"{DB_COMPONENT_NAME}.user")
+            self.password = get_config_value(config, f"{DB_COMPONENT_NAME}.password")
+        except KeyError as ex:
+            raise ValueError(
+                f"Credentials file '{credentials_file_path}' does not contain key '{ex}'."
+            )
+
+    def load_credentials_from_env(self):
+        """
+        :raise ValueError if any expected environment variable is not set.
```

Comment:
```suggestion
        :raises ValueError: if any expected environment variable is not set.
```

Seems like that's the correct format according to: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format

Reply:
they might be the same according to this doc:
https://www.sphinx-doc.org/en/master/usage/domains/python.html#:~:text=raises%2C%20raise%2C%20except%2C%20exception%3A%20That%20(and%20when)%20a%20specific%20exception%20is%20raised.

Reply:
Sure, we can use `raise`. (Btw, the other difference is putting another colon after the type.)